integration_test/reporting_error_index/reporting_error_index_test.go

```diff
@@ -0,0 +1,814 @@
+package reportingfailedmessages_test
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/jobsdb"
+
+	"github.com/rudderlabs/rudder-server/processor/transformer"
+
+	"github.com/rudderlabs/rudder-server/testhelper/destination"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	kithttputil "github.com/rudderlabs/rudder-go-kit/httputil"
+	kithelper "github.com/rudderlabs/rudder-go-kit/testhelper"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/rand"
+	"github.com/rudderlabs/rudder-server/runner"
+	"github.com/rudderlabs/rudder-server/testhelper/backendconfigtest"
+	"github.com/rudderlabs/rudder-server/testhelper/health"
+	"github.com/rudderlabs/rudder-server/testhelper/transformertest"
+
+	_ "github.com/marcboeker/go-duckdb"
+)
+
+func TestReportingErrorIndex(t *testing.T) {
+	t.Run("Events failed during tracking plan validation stage", func(t *testing.T) {
+		config.Reset()
+		defer config.Reset()
+
+		bcServer := backendconfigtest.NewBuilder().
+			WithWorkspaceConfig(
+				backendconfigtest.NewConfigBuilder().
+					WithSource(
+						backendconfigtest.NewSourceBuilder().
+							WithID("source-1").
+							WithWriteKey("writekey-1").
+							WithTrackingPlan("trackingplan-1", 1).
+							WithConnection(
+								backendconfigtest.NewDestinationBuilder("WEBHOOK").
+									WithID("destination-1").
+									Build()).
+							Build()).
+					Build()).
+			Build()
+		defer bcServer.Close()
+
+		trServer := transformertest.NewBuilder().
+			WithTrackingPlanHandler(
+				transformertest.ViolationErrorTransformerHandler(
+					http.StatusBadRequest,
+					"tracking plan validation failed",
+					[]transformer.ValidationError{{Type: "Datatype-Mismatch", Message: "must be number"}},
+				),
+			).
+			Build()
+		defer trServer.Close()
+
+		pool, err := dockertest.NewPool("")
+		require.NoError(t, err)
+
+		postgresContainer, err := resource.SetupPostgres(pool, t)
+		require.NoError(t, err)
+		minioResource, err := destination.SetupMINIO(pool, t)
```

Comment:
We have recently introduced a minio resource in rudder-go-kit. We should start using [this](https://github.com/rudderlabs/rudder-go-kit/blob/main/testhelper/docker/resource/minio.go).

Reply:
Yeah, I was waiting for `rudder-go-kit` to be updated to `1.16.1`

Reply:
you can do the update here